address_book.py

```diff
@@ -10,57 +10,68 @@ def __init__(self, name, email_address, phone_number):
 
     def show(self):
         """Shows details."""
-        print(f'Name = {self.name}, Email Address = {self.email_address}, Phone number = {self.phone_number}'
+        print(
+            f'Name = {self.name}, Email Address = {self.email_address}, Phone number = {self.phone_number}')
 
 
 class AddressBook:
-    def __init__(self):
-        self.contacts = {}
+    '''Write a name for your file as the argument for the AddressBook you are instantiating. 
+       This will be the file where the contacts are saved. .json will be added to the filename automatically.'''
+    # dont need to duplicate name
+
+    def __init__(self, contactsfile):
+        # This is where the data will be saved
+        self.contactsfile = contactsfile
+        self._contacts = {}
+        # if the file already exists, load the data in into the _contacts dict by iterating through the file
+        if Path(self.contactsfile).exists == True:
+            with open(self.contactsfile) as f:
+                data = json.load(f)
+
+            for name, details in data.items():
+                self._contacts[name] = Contact(
+                    name, details['email_address'], details['phone_number'])
+
+        elif Path(self.contactsfile).exists == False:
+            open(self.contactsfile)
 
     def add_contact(self, contact, object):
         """add contact to address book, contact is the key you will use to find the contact and
          the value is the contact object"""
-        self.contacts[contact] = object
+        self._contacts[contact] = vars(object)
```

Comment:
What is going on here? What is contact? Should the arg contact be a string? Why not just use contact.name as the key and pass in a single contact obj. Ie

`
def add_contact(self, contact):
        """add contact to address book contact is an instance of Contact"""
        self._contacts[contact.name] = contact
`